src/classes/BDI_MappingServiceAdvanced.cls

```diff
@@ -66,30 +66,35 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
 
         String sourceObject = DataImport__c.SObjectType.getDescribe().getName();
         for (String dataImportObjectName : fieldMappingsByObjMappingDevName.keySet()) {
-
-            String targetObject =
-                    objectMappingByDevName.get(dataImportObjectName).Object_API_Name;
-            List<BDI_FieldMapping> fieldMappingsForThisDataImportObject =
-                    fieldMappingsByObjMappingDevName.get(dataImportObjectName);
-
-            for (BDI_FieldMapping dataImportFieldMapping :
-                    fieldMappingsForThisDataImportObject) {
-
-                String sourceField = dataImportFieldMapping.Source_Field_API_Name;
-
-                SObjectField sourceSObjectField =
-                        UTIL_Describe.getFieldDescribe(
-                                sourceObject, sourceField).getSobjectField();
-
-                String targetField = dataImportFieldMapping.Target_Field_API_Name;
-
-                if (targetFieldsBySourceField.keySet().contains(sourceSObjectField)) {
-                    targetFieldsBySourceField.get(
-                            sourceSObjectField).addTargetField(targetObject, targetField);
-                } else {
-                    BDI_TargetFields targetFields = new BDI_TargetFields();
-                    targetFields.addTargetField(targetObject, targetField);
-                    targetFieldsBySourceField.put(sourceSObjectField, targetFields);
+            BDI_ObjectMapping targetObjectMapping = objectMappingByDevName.get(dataImportObjectName);
+            if(targetObjectMapping != null) {
```

Comment:
the `if` block is the main change here - if the targetObjectMapping does not exist we need to skip it, because the target SObject was likely deleted.